lib/spack/spack/solver/asp.py

```diff
@@ -4186,6 +4187,97 @@ def execute_explicit_splices(self):
         return specs
 
 
+def _attach_python_to_external(
+    dependent_package, extendee_spec: Optional[spack.spec.Spec] = None
+) -> None:
+    """
+    Ensure all external python packages have a python dependency
+
+    If another package in the DAG depends on python, we use that
+    python for the dependency of the external. If not, we assume
+    that the external PythonPackage is installed into the same
+    directory as the python it depends on.
+    """
+    # TODO: Include this in the solve, rather than instantiating post-concretization
+    if "python" not in dependent_package.spec:
+        if extendee_spec:
+            python = extendee_spec
+        else:
```

Comment:
The elif was removed on account of https://github.com/spack/spack/pull/34202/files#r1070191430  no longer applying.